### PR TITLE
refactor(test_ode.py): 1st_exact examples

### PR DESCRIPTION
**Pull request by Mohitbalwani26**
#### References to other Issues or PRs

SEE #18377

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/solvers/ode/tests/test_single.py [hunk 1]
```diff
@@ -1284,6 +1289,48 @@ def _get_examples_ode_sol_separable():
...
+    '1st_exact_01': {
+        'eq': sin(x)*cos(f(x)) + cos(x)*sin(f(x))*f(x).diff(x),
+        'sol': [Eq(f(x), -acos(C1/cos(x)) + 2*pi), Eq(f(x), acos(C1/cos(x)))],
+        'slow': True,
+    },
+
+    '1st_exact_02': {
+        'eq': (2*x*f(x) + 1)/f(x) + (f(x) - x)/f(x)**2*f(x).diff(x),
+        'sol': [Eq(f(x), exp(C1 - x**2 + LambertW(-x*exp(-C1 + x**2))))],
+        'slow': True,
```
Comment (Mohitbalwani26) [inline sympy/solvers/ode/tests/test_single.py:1308]: This solution fails when tested in `checkodesol` and when we keep `sol=Eq(log(f(x)) + x/f(x) + x**2, C1)` it passes but test will still fail as dsolve output doesn't match with this.

File: sympy/solvers/ode/tests/test_single.py [hunk 2]
```diff
@@ -1284,6 +1289,48 @@ def _get_examples_ode_sol_separable():
...
+    '1st_exact_04': {
+        'eq': cos(f(x)) - (x*sin(f(x)) - f(x)**2)*f(x).diff(x),
+        'sol': [Eq(x*cos(f(x)) + f(x)**3/3, C1)],
+        'slow': True,
+    },
+
+    '1st_exact_05': {
+        'eq': 2*x*f(x) + (x**2 + f(x)**2)*f(x).diff(x),
+        'sol': [Eq(x**2*f(x) + f(x)**3/3, C1)],
+        # 'sol':[Eq(f(x), 2**Rational(1, 3)*(x**2/(-3*C1 + sqrt(9*C1**2 + 4*x**6))**Rational(1, 3) - 2**Rational(1, 3)*(-3*C1 + sqrt(9*C1**2 + 4*x**6))**Rational(1, 3)/2)), Eq(f(x), 2**Rational(1, 3)*(2*x**2/((-1 - sqrt(3)*I)*(-3*C1 + sqrt(9*C1**2 + 4*x**6))**Rational(1, 3)) + 2**Rational(1, 3)*(-3*C1 + sqrt(9*C1**2 + 4*x**6))**Rational(1, 3)/4 + 2**Rational(1, 3)*sqrt(3)*I*(-3*C1 + sqrt(9*C1**2 + 4*x**6))**Rational(1, 3)/4)),
```
Comment (Mohitbalwani26) [inline sympy/solvers/ode/tests/test_single.py:1326]: currently after refactoring the output of dsolve gets simplified but earlier for comparing solution with dsolve it was tested as: `assert dsolve(eq5, hint='1st_exact', simplify=False) == sol5`

File: sympy/solvers/ode/tests/test_single.py (hunk 1 above)
Comment (oscarbenjamin) [reply sympy/solvers/ode/tests/test_single.py:1308]: Is there a way to note that checkodesol is xfail for this test?

Comment (Mohitbalwani26) [reply sympy/solvers/ode/tests/test_single.py:1308]: No, currently xfail only stores the hints which it fails for.

Comment (Mohitbalwani26) [reply sympy/solvers/ode/tests/test_single.py:1308]: should there be way for getting xfail on checkodesol?

File: sympy/solvers/ode/tests/test_single.py (hunk 2 above)
Comment (Mohitbalwani26) [reply sympy/solvers/ode/tests/test_single.py:1326]: Any suggestion for this case as we need to explicitly pass the simplify argument?

File: sympy/solvers/ode/tests/test_single.py (hunk 1 above)
Comment (oscarbenjamin) [reply sympy/solvers/ode/tests/test_single.py:1308]: Yes I think so.

File: sympy/solvers/ode/tests/test_single.py (hunk 2 above)
Comment (oscarbenjamin) [reply sympy/solvers/ode/tests/test_single.py:1326]: There should be a way to specify simplify=False.

Comment (Mohitbalwani26) [reply sympy/solvers/ode/tests/test_single.py:1326]: So should I keep simplify_flag as a key in the example?

Comment (oscarbenjamin) [reply sympy/solvers/ode/tests/test_single.py:1326]: Yes, I think so.

**Comment by Mohitbalwani26** [comment]
@oscarbenjamin Does this approach look good now?
I don't want this PR to get merged now as I will be refactoring more examples in it.

**Comment by oscarbenjamin** [comment]
Yes, it looks good to me

File: sympy/solvers/ode/tests/test_ode.py [hunk 3]
```diff
@@ -813,35 +813,6 @@ def test_old_ode_tests():
...
-    # FIXME: assert checkodesol(eq2, sol2, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq2, sol2b, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq3, sol3, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq4, sol4, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq5, sol5, order=1, solve_for_func=False)[0]
-
-
 @slow
 @XFAIL
 def test_1st_exact2_broken():
```
Comment (Mohitbalwani26) [inline sympy/solvers/ode/tests/test_ode.py:818]: @oscarbenjamin this test is too slow for travis and it is splitted in to two functions one for verifying dsolve_sol and another for checking ode_sol. so how should i refactor this?

Comment (oscarbenjamin) [reply sympy/solvers/ode/tests/test_ode.py:818]: We need a way to mark some examples as being too slow either in dsolve or checkodesol.

Comment (Mohitbalwani26) [reply sympy/solvers/ode/tests/test_ode.py:818]: The cases in which dsolve hangs when tested with other hints, I marked them as XFAIL. But in this case the main hint only is taking time, so should I add another key as dsolve_slow and checkodesol_slow?

Comment (oscarbenjamin) [reply sympy/solvers/ode/tests/test_ode.py:818]: Yes, something like that

Comment (Mohitbalwani26) [reply sympy/solvers/ode/tests/test_ode.py:818]: so,if one of these flag is set to be  True then should the test be skipped on Travis just as written in `test_1st_exact2_broken()`?

Comment (oscarbenjamin) [reply sympy/solvers/ode/tests/test_ode.py:818]: If dsolve is too slow then testing dsolve should be skipped. If checkodesol is too slow then testing checkodesol should be skipped.

**Comment by Mohitbalwani26** [comment]
@oscarbenjamin can you review this now? I have refactored all the examples of 1st_exact hint.

**Comment by oscarbenjamin** [comment]
This looks good

**Pull request merged**